packages/preferences/src/browser/util/preference-event-service.ts

```diff
@@ -0,0 +1,27 @@
+/********************************************************************************
+ * Copyright (C) 2020 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { injectable } from 'inversify';
+import { Emitter } from '@theia/core/lib/common/event';
+import { Preference } from './preference-types';
+
+@injectable()
+export class PreferencesEventService {
```

Comment:
It does not look good. The should be an object which keeps some state, this object expose events which indicate that state is changed, emitters should not be exposed and shared.

Reply:
Let's skip it for now. It will be another serious refactoring to encapsulate everything properly.